src/TS3Connection.ts

```diff
@@ -514,20 +524,27 @@ export class TS3Listener extends events.EventEmitter {
         // do not write leads of members which hide their roles
         const writeToDB: boolean = !(dmember && dmember.roles.cache.find(r => getConfig().get().achievements.ignoring_roles.includes(r.name)));
         if (writeToDB) {
-            log("debug", "Writing raid information to database.");
+            LOG.debug("Writing raid information to database.");
             if (commander.getRaidStart() === undefined) {
-                log("error", `Wanted to write raid for commander ${dmember.displayName} during tag-down, but no raid start was stored.`);
+                LOG.error(`Wanted to write raid for commander ${dmember.displayName} ` +
+                    "during tag-down, but no raid start was stored.");
             } else {
-                this.botgartClient.tsLeadRepository.addLead(registration.gw2account, <moment.Moment>commander.getRaidStart(), moment.utc(), commander.getTS3Channel());
+                this.botgartClient.tsLeadRepository.addLead(
+                    registration.gw2account,
+                    <moment.Moment>commander.getRaidStart(),
+                    moment.utc(),
+                    commander.getTS3Channel());
             }
-            log("debug", "Done writing to database.");
+            LOG.debug("Done writing to database.");
         }
     }
 
     private async tagUpAssignRole(g: discord.Guild, commander: Commander) {
         const crole = g.roles.cache.find(r => r.name === this.commanderRole);
         if (crole && commander.getDiscordMember()) {
-            await commander.getDiscordMember()?.roles.add(crole);
+            await commander.getDiscordMember()?.roles.add(crole)
+                .catch(e => LOG.warn(`Could not remove role '${this.commanderRole}' from ` +
```

Comment:
same as above: I would recommend using either `await` + `try-catch` or constently using the promise-syntax where `then()` and `catch()` are chained to pending Promises.

Reply:
TBH, in some cases if I just find this just much nicer to read and much shorter.
A try-catch would not only catch errors from the Promise, but also from the method/property chain beforehand, which is not wanted here.

Reply:
I don't see how that is the case.

```javascript
await commander.getDiscordMember()?.roles.add(crole).catch(e => LOG.warn(e));
```

should be the same as 

```javascript
try {
  await commander.getDiscordMember()?.roles.add(crole);
} catch(e) {
  LOG.warn(e);
}
```
as `getDiscordMember()` hopefully shouldn't throw any exceptions.

Although... reading it again I am actually not sure what is being `await`ed here... or rather if the code does what you intend it to do.
According to the [ECMA standard](https://www.ecma-international.org/wp-content/uploads/ECMA-262-10th-edition-June-2019.pdf) `await` behaves as a unary operator which gives it precedence over binary operators (ie `.`). I currently have no means of trying this, but it reads a lot like you are `await`ing `commander`, which should be a non-thenable already and then the chaining just runs its course.
So... what did you actually intend to `await` here?

Reply:
I await for the role to be added ( or not if it fails ), none the less the code should continue here and should not abort.

Reply:
Alright, I just checked the [operator precedence table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence) and member access does have a higher precedence than `await` – you're right.